tests/contract-playground/src/ArithmeticUnderflowOverflow.sol

```diff
@@ -0,0 +1,122 @@
+// SPDX-License-Identifier: MIT
+pragma solidity <0.8.0;
+ 
+contract ArithmeticUnderflowOverflow {
+    uint256 public counter;
+
+    // bad
+    function add(uint256 num1, uint256 num2) public pure returns (uint256 result) {
+      result = num1 + num2;
+    }
+
+    // bad
+    function sub(uint256 num1, uint256 num2) public pure returns (uint256 result) {
+      result = num1 - num2;
+    }
+
+    // bad
+    function mul(uint256 num1, uint256 num2) public pure returns (uint256 result) {
+      result = num1 * num2;
+    }
+
+    // bad
+    function div(uint256 num1, uint256 num2) public pure returns (uint256 result) {
+      result = num1 / num2;
+    }
+
+    // bad
+    function incrementCounter() public {
+      counter += 1;
+    }
+
+    // bad
+    function decrementCounter() public {
+      counter += 1;
+    }
+
+    // bad
+    function incrementCounter2() public {
+      counter++;
+    }
+
+    // bad
+    function decrementCounter2() public {
+      counter--;
+    }
+
+    // good
```

Comment:
This currently is captured by the detector, even though it is a false-positive.

This function is `// good` because there is a check within the function before the arithmetic operation occurs to ensure no underflow happens. This is what SafeMath does under the hood.

There is a way we can alter this detector so that it doesn't capture this as a false-positive. The trick is to look for:
* arithmetic operations on identifiers (which the detector almost does already, it just doesn't track the identifiers yet)
* other checks that occur within the same function, on the same identifiers.

If both are true, then the detector probably shouldn't capture this.

Reply:
@alexroan Got it. I'll tackle these.